### PR TITLE
added Iso19115-2 Spatial processing to dcatus

### PR DESCRIPTION
**Pull request by Jin-Sun-tts**
related to https://github.com/GSA/data.gov/issues/4893

- added spatial processing in the resourceInfo->extent->geographic_extent module
- added test

### CODE REVIEW

File: lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_extent.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module Extent
+               @@exExtentXPath = 'gmd:EX_Extent'
+               @@descXPath = 'gmd:description//gco:CharacterString'
+               @@geoElemXPath = 'gmd:geographicElement'
+               @@temporalElemXPath = 'gmd:temporalElement'
+               @@verticalElemXPath = 'gmd:verticalElement'
```
Comment (rshewitt) [inline lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_extent.rb:16]: remove these

**Comment by rshewitt** [comment]
- bounding box test module is missing. you need to include a reader test verifying data was read and placed into the internal metadata object as intended. you're also missing the error message test. you can consolidate them all into 1 test instead of 4 that's fine.
- geographic extent test module is missing. add it.

File: test/readers/iso19115_2/tc_iso19115_2_extent.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      xIn = xDoc.xpath('.//gmd:extent')[0]
+      hResponse = Marshal.load(Marshal.dump(@@hResponseObj))
+      hDictionary = @@nameSpace.unpack(xIn, hResponse)
+
+      refute_empty hDictionary
+      assert hDictionary.instance_of? Hash
+
+      hGeoExtent = hDictionary[:geographicExtents][0]
+      refute_empty hGeoExtent[:boundingBox]
+      assert_equal("true", hGeoExtent[:containsData])
```
Comment (rshewitt) [inline test/readers/iso19115_2/tc_iso19115_2_extent.rb:25]: just check that `hDictionary[:geographicExtents]` is of particular size (i.e. `assert_equal(1, hDictionary[:geographicExtents].size)` . add a geo extent test module and have it be responsible for its own content. typically we want the child to be responsible for asserting its own content

Comment (rshewitt) [reply test/readers/iso19115_2/tc_iso19115_2_extent.rb:25]: a module which contains other modules ( e.g. `extent` contains an array of `:geographicExtents` ) should simply check the size of that array not the content of the elements in the array itself

Comment (Jin-Sun-tts) [reply test/readers/iso19115_2/tc_iso19115_2_extent.rb]: added all above tests, please review again. Thanks!

File: test/readers/iso19115_2/tc_iso19115_2_extent.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      xDoc = TestReaderIso191152Parent.get_xml('iso19115-2.xml')
+      TestReaderIso191152Parent.set_xdoc(xDoc)
+
+      xIn = xDoc.xpath('.//gmd:extent')
+      hResponse = Marshal.load(Marshal.dump(@@hResponseObj))
+      hDictionary = @@nameSpace.unpack(xIn, hResponse)
+
+      refute_empty hDictionary
+      assert hDictionary.instance_of? Hash
+      if hDictionary.key?(:geographicExtents)
```
Comment (rshewitt) [inline test/readers/iso19115_2/tc_iso19115_2_extent.rb:22]: `extent` mbjson hashes always have `geographicExtents` as a key so this condition shouldn't be necessary

Comment (rshewitt) [reply test/readers/iso19115_2/tc_iso19115_2_extent.rb:22]: if you can fix this then i'll approve. good work!

**Pull request merged**